### PR TITLE
Paraswap: Helper query methods for claimer

### PR DESCRIPTION
**Pull request by facuspagnuolo**

### CODE REVIEW

**Review by ragonzal — APPROVED** [review]
Looking good, `totalBalance()` function will reduce the bot queries
Thanks!

File: packages/base/contracts/actions/RelayedAction.sol [hunk 1]
```diff
@@ -144,10 +144,9 @@ abstract contract RelayedAction is BaseAction {
     /**
      * @dev Internal function to fetch the paying gas token rate from the Smart Vault's price oracle
      */
-    function _getPayingGasTokenPrice(address payingToken) private view returns (uint256) {
-        address wrappedNativeToken = smartVault.wrappedNativeToken();
-        bool isUsingNativeToken = payingToken == Denominations.NATIVE_TOKEN || payingToken == wrappedNativeToken;
-        return isUsingNativeToken ? FixedPoint.ONE : smartVault.getPrice(wrappedNativeToken, payingToken);
+    function _getPayingGasTokenPrice(address token) private view returns (uint256) {
+        bool isUsingNativeToken = _isWrappedOrNativeToken(token);
```
Comment (dmf7z) [inline packages/base/contracts/actions/RelayedAction.sol:148]: do we need this bool?
`return _isWrappedOrNativeToken(token) ? FixedPoint.ONE : smartVault.getPrice(smartVault.wrappedNativeToken(), token);`

Comment (facuspagnuolo) [reply packages/base/contracts/actions/RelayedAction.sol:148]: the linter makes it uglier, declaring the variable does not make anything worse, the stack will need to require to memory anyway

Comment (dmf7z) [reply packages/base/contracts/actions/RelayedAction.sol:148]: 👍

**Review by dmf7z — APPROVED** [review]
Looks good 👍

**Pull request merged**